File: src/com/rian/difficultycalculator/attributes/DifficultSlider.java
package com.rian.difficultycalculator.attributes;

/**
 * Represents a slider that is considered difficult.
 */
public final class DifficultSlider {
    /**
     * The index of the slider in the beatmap.
     */
    public final int index;

    /**
     * The difficulty rating of this slider compared to other sliders, based on the velocity of the slider.
     * <br><br>
     * A value closer to 1 indicates that this slider is more difficult compared to most sliders.
     * <br><br>
     * A value closer to 0 indicates that this slider is easier compared to most sliders.
     */
    public double difficultyRating;

    /**
     * @param index The index of the slider in the beatmap.
     * @param difficultyRating The difficulty rating of this slider compared to other sliders, based on the velocity of the slider.
     */
    public DifficultSlider(int index, double difficultyRating) {
        this.index = index;
        this.difficultyRating = difficultyRating;
    }
}

File: src/com/rian/osu/difficulty/attributes/DifficultSlider.kt
package com.rian.osu.difficulty.attributes

import com.rian.osu.beatmap.Beatmap
import com.rian.osu.beatmap.hitobject.Slider

/**
 * Represents a [Slider] that is considered difficult.
 */
data class DifficultSlider(
    /**
     * The index of the [Slider] in the [Beatmap].
     */
    @JvmField
    val index: Int,

    /**
     * The difficulty rating of this [Slider] compared to other [Slider]s, based on the velocity of the [Slider].
     *
     * A value closer to 1 indicates that this [Slider] is more difficult compared to most [Slider]s.
     *
     * A value closer to 0 indicates that this [Slider] is easier compared to most [Slider]s.
     */
    @JvmField
    val difficultyRating: Double
)